content/pt-br/blog/_posts/2023-03-10-image-registry-change.md

```diff
@@ -0,0 +1,158 @@
+
+## Eu fui afetado, o que devo fazer?
+Para usuários afetados que usam um ambiente restrito, a melhor opção é copiar as imagens necessárias
+para um registro privado ou configurar um cache de pull-through nos seu registry.
+
+Existem várias ferramentas para copiar imagens entre registries. [Crane](https://github.com/google/go-containerregistry/blob/main/cmd/crane/doc/crane_copy.md) é uma dessas ferramentas, as imagens podem ser copiadas para um registry privado com `crane copy SRC DST`. Também existem ferramentas específicas de fornecedores, como o [gcrane](https://cloud.google.com/container-registry/docs/migrate-external-containers#copy) que faz uma função similar mas simplificada para a plataform da Google.
+
+
+## Como eu posso encontrar quais imagens estão usando o registry antigo, e corrigir elas?
+
+**Opção 1**: Veja esse comando do kubectl no [blog post anterior](https://kubernetes.io/blog/2023/02/06/k8s-gcr-io-freeze-announcement/#what-s-nex):
+
+```
+kubectl get pods --all-namespaces -o jsonpath="{.items[*].spec.containers[*].image}" |\
+tr -s '[[:space:]]' '\n' |\
+sort |\
+uniq -c
+```
+
+**Opção 2**: Um plugin do `kubectl` [krew]((https://krew.sigs.k8s.io/) chamado [`community-images`](https://github.com/kubernetes-sigs/community-images#kubectl-community-images) foi desenvolvido para escanear e reportar qualquer
+imagem usando o endpoint k8s.gcr.io.
+
+Se você tem o krew instalado, pode instalar o plugin com:
+
+```
+kubectl krew install community-images
+```
+
+E gerar um relatório com:
+
+```
+kubectl community-images
+```
+
+Para métodos alternativos de instalação ou exemplos de saída, veja o repositório [kubernetes-sigs/community-images](https://github.com/kubernetes-sigs/community-image).
+
+**Opção 3**: Se você não tem acesso diretamente ao cluster, ou gerencia muitos clusters - o melhor
+caminho é executar uma busca sobre seus manifestos e charts por _"k8s.gcr.io"_.
+
+**Opção 4**: Se você quer prevenir a execução de imagens a partir do k8s.gcr.io no seu cluster, existem políticas
+para [Gatekeeper](https://open-policy-agent.github.io/gatekeeper-library/website/) e
+[Kyverno](https://kyverno.io/) disponíveis no [repositório de melhores práticas para AWS EKS](https://github.com/aws/aws-eks-best-practices/tree/master/policies/k8s-registry-deprecation) que vão bloquear o pull das imagens. Você
+pode usar essas políticas de terceiros com qualquer cluster Kubernetes.
+
+**Opção 5**: Como a **ÚLTIMA** opção, você pode usar um [webhook de admissão mutating](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks) para alterar o
+endereço da imagem dinamicamente. Isso deve ser feito apenas como uma medida paliativa enquanto os
+seus manifestos são atualizados. Você pode encontrar um webhook de admissão mutating e uma política do 
+kyverno (de terceiros) em [k8s-gcr-quickfix](https://github.com/abstractinfrastructure/k8s-gcr-quickfix).
```

Comment:
Some suggestions here:
* Omit "mutating" completely in "webhook de admissão", since "alterar o endereço" already delivers the idea of mutation.
* Use capital K in Kyverno.